client/src/components/Navbar/Navbar.jsx

```diff
@@ -1,15 +1,55 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import Link from './Link';
 import Logo from '../Logo';
+import Hamburger from './Hamburger';
 
 const Navbar = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [height, setHeight] = useState(0);
+  const firstUpdate = useRef(true);
+
+  const handleClick = () => {
+    setIsOpen((isOpen) => !isOpen);
+  };
+
+  useEffect(() => {
+    // handling the componentDidMount
+    if (firstUpdate.current) {
+      firstUpdate.current = false;
+      return;
+    }
+
+    // handling the componentDidUpdate
+    const maxHeight = 120;
+
+    new Promise((resolve, reject) => {
+      let percent = 0;
+
+      const step = () => {
+        percent += 2.5;
```

Comment:
Ta funkcja wygląda bliźniaczo do tej z PR1 - albo mi się wydaje.
Jeśli jest bliźniacza to warto może to wydzielić i sparametryzować w wywołaniu

I oczywiście magic numbers :)